PhpAmqpLib/Channel/AMQPChannel.php

```diff
@@ -844,13 +844,19 @@ public function basic_nack($delivery_tag, $multiple = false, $requeue = false)
      *
      * @param string $consumer_tag
      * @param bool $nowait
+     * @param bool $noreturn
      * @return mixed
      */
-    public function basic_cancel($consumer_tag, $nowait = false)
+    public function basic_cancel($consumer_tag, $nowait = false, $noreturn = false)
```

Comment:
I will try and make a demo when I get a chance. I'll remove it if it moves the pull request along. I assumed the spec specific functionality was in the Protocol folder. There seemed to be a few things already in here that didn't comply. 

No wait doesn't expect a message response from the broker and no return doesn't wait for one. It's different in that I still need a response from the broker but I do not want to wait (block) in the code for one. Getting signals to work in this pull request uncovered the issue. When I send it a signal to stop consuming and close the code wouldn't exit. Signals can be sent at any point and after dealing with them it returns back to the last place it left off.